test/parallel/test-repl-no-sab.js

```diff
@@ -0,0 +1,11 @@
+// Flags: --no-harmony-sharedarraybuffer
+
+'use strict';
+
+const assert = require('assert');
+const repl = require('repl');
+
+// Regression test for https://github.com/nodejs/node/issues/39717.
+assert.doesNotThrow(() => {
+  repl.start();
+});
```

Comment:
`assert.doesNotThrow()` does not provide any benefit over just running the code without being wrapped and is therefore prohibited with a linter. :-)

I would also change the test to start a worker instead of the REPL as it is the part that uses the `SharedArrayBuffer`.